### ISSUE TITLE
JOSS Review (@mkearney): Export status id_str

### ISSUE THREAD

**Issue by mkearney** [IUNetSci/BotSlayer-CE#23]
* [x] The tool does not currently output a useful identifier of statuses. The exported output includes bot scores, screen names, status texts, timestamps, etc., but Twitter's stream and REST APIs [forcefully] return a lot more data than that. I don't think BotSlayer should include/return all of this information for the user, but it should at least return the sufficient information for researchers to lookup the rest via requests to the *lookup/statuses* endpoint of Twitter's REST API. **The exported output should therefore return the status `id_str` value**.

**Comment by marcmccarty** [comment]
Would you suggest some kind of non-comma-delimited list for this? 

Each exported entity is supposed to represent all tweets that the BotSlayer-CE instance gathered, so there could either be a list of statuses as the last column, potentially, or otherwise many copies of the same entity with the only difference being the last column, `id_str`.

**Comment by phui** [comment]
@mkearney 

Thank you very much for providing your review for our tool! I would like to let you know we are working on items in your review (and those in your colleague's review).

Exporting tweet IDs/user IDs is one of the things that get tricky because of Terms of Service (TOS).

When the owner of the machine shares the DNS of the web interface, but not the host machine, to external users, bulk exporting of tweet IDs/User IDs may lead to a violation of Twitter TOS (F2a. at https://developer.twitter.com/en/developer-terms/agreement-and-policy.html), that the owner of the dataset shares too much content with other people.

If the user or the group of users owns the machine that collects the data, instead of bulk exporting tweet IDs/User IDs for post-analysis via web interface, the `PostgreSQL` database connection on the host machine enables collection of tweets on simple filtering condition.
The tool instead provides an export function for results of more complicated aggregations.

Since you have brought up the issue, we think the export functionality is crucial to users. Especially for journalist we should provide such function, although it may be locked behind authentication (to prove ownership of the dataset). Technically it is feasible, but it will require more design thoughts. It will enter the feature list for the next version release.

Does this answer your question?

**Comment by filmenczer** [comment]
These are good points. Since the sharing would be done by the user, the responsibility would be technically theirs. Perhaps we could just provide a warning that they need to comply with Twitter terms and link to ToS when they export data that includes tweet IDs.

**Comment by mkearney** [comment]
> If you provide Twitter Content to third parties, including downloadable datasets of Twitter Content or an API that returns Twitter Content, you will only distribute or allow download of Tweet IDs, Direct Message IDs, and/or User IDs.

That’s from the linked page. If anything providing *only* IDs may be the only way to not violate TOS?

**Comment by filmenczer** [comment]
Let me try to clarify our point. We (the developers) do not provide Twitter Content to third parties. BotSlayer users get the content directly from the Twitter API, using their own Twitter app (developer) keys. 

If a user then provides data collected from Twitter via BotSlayer to third parties, they have to comply with Twitter terms. We would like to make it less likely that they inadvertently violate Twitter terms. The data that they can currently export is aggregated statistical data, not raw Twitter content. That should be safe.

If we make it easy for users to share actual Twitter content, then you are right, it should be tweet and user IDs only. But Twitter terms also impose limits on how many IDs can be shared, unless for research purposes -- this is what @phui was referring to.

Therefore we agree with you, but if we make it easy to export IDs (which are actual Twitter Content), we should alert the users to minimize the chances that they inadvertently violate Twitter terms. This was the sense of my previous comment.

**Comment by mkearney** [comment]
@filmenczer That makes a lot more sense–and it's what I figured your logic was–but in that case it seems like you all may be **overly** protective. I'll think about this more and try to type a more coherent response later today, but I think our goals are the same [and you all have already done a great job with this], so I would consider my issues to be more like 'requests' (it'd be nice to allow researchers easy access to status/user IDs) than 'demands' (these are not deal breakers to me recommending for publication).

**Comment by mkearney** [comment]
I think I was probably over-thinking this. I think the non-comma (space or +) delimited string of ID values, like @marcmccarty said, would be perfect!

**Comment by filmenczer** [comment]
We discussed the tradeoff between ease of use and risks of inadvertently violating Twitter policies. As a compromise, we are not exporting IDs via the dashboard interface, but we added instructions in the README about how to access and export the data in the backend by accessing the database directly. We also added a pointer to Twitter policies on resharing data with third parties. 

This solution will (1) make it more likely that users are aware of Twitter terms before they share data, and (2) make it much easier/faster for users to access raw Twitter content without having to rehydrate the data via the Twitter API.

We believe this should address this issue.
